lib/galaxy/datatypes/sequence.py

```diff
@@ -818,9 +821,25 @@ class FastqSanger(Fastq):
     @staticmethod
     def quality_check(lines):
         """Presuming lines are lines from a fastq file, return True if the qualities are compatible with sanger encoding"""
+        is_ambiguous = True
+        within_typical_upper_bound = True
         for line in islice(lines, 3, None, 4):
-            if not all(_ >= '!' and _ <= 'M' for _ in line[0]) or ' ' in line:
```

Comment:
So what was the highest value ? wikipedia says 45 for moleculo reads. I guess we could reasonably go until 50 (`S`, right?) and still wouldn't falsely sniff solexa and illumina 1.3-1.5 (because that should correspond to 24, and the majority of reads should have higher quality bases, falling outside of the range).

Reply:
The highest in our training dataset is 46 (`O`). I couldn't find any information about currently used upper limits on standard Illumina platforms (all official Illumina material I found is very old, and I have no idea how up-to-date the Wikipedia article is).
`S` would actually correspond only to 19; Solexa scores start at 59 (`;`), but that would be a score of -5 (according to Wikipedia).
Anyway my idea was to leave the limit at `M` since then there would not be any change in Galaxy's sniffing outcome - except for cases that were clearly wrong detections before (those that must be fastqsanger because they contain ASCII values < 59).
If you think we should raise it to `S` or whatever other limit we can surely do that, too.

Reply:
I think we should only raise it to `S`, yes (and thanks for the `-5`, forgot that the scale actually starts there). This is far more lightweight and easier to understand, plus the current logic would fail for reads with high quality bases (i.e `within_typical_upper_bound ` is `False`) that don't also have a low quality base (`is_ambiguous ` is `True`). You'd assume that as sequencing technology improves we'd also have fewer low quality bases, so I don't think this is a good heuristic.

Reply:
Ok, so just one last defense of the heuristics before I reduce this to the minimal change:
> plus the current logic would fail for reads with high quality bases (i.e `within_typical_upper_bound ` is `False`) that don't also have a low quality base (`is_ambiguous ` is `True`).

in which case it would behave exactly like the *status quo*! As the title says the proposal would have made detection of fastqsanger *more* robust, not less. It would have used proof for fastqsanger if possible and have guessed in ambiguous situations.

Anyway, lets just move on and make the training material work.